activesupport/lib/active_support/cache.rb

```diff
@@ -18,6 +18,65 @@ module Cache
     autoload :NullStore,        "active_support/cache/null_store"
     autoload :RedisCacheStore,  "active_support/cache/redis_cache_store"
 
+
+    class KeyMaster
+      attr_reader :cache_key, :cache_version
+
+      def initialize(key = nil)
+        @cache_key = String.new
```

Comment:
Do you want to create a string with ASCII encoding or UTF8 encoding?

Reply:
Unsure, we probably want to handle UTF 8 cache keys.

Reply:
Yeah, so I think we should be using `+""`. String.new changes the encoding of the string.

```
>> String.new.encoding
=> #<Encoding:ASCII-8BIT>
>> "".encoding
=> #<Encoding:UTF-8>
```